src/relay/pass/pattern_util.h

```diff
@@ -373,6 +374,26 @@ inline Expr Copy(Expr data) {
 }
 
 
+inline Expr Where(const Expr& condition, const Expr& x, const Expr& y) {
```

Comment:
s/inline/static inline

Reply:
Replace inline with static inline.

Reply:
Will skip this for now. There are other functions in this file, which need to be static inline. Will send out a separate patch for the whole file, if need be.

Reply:
I would encourage you to do that with this PR itself because of the accidents missing this can cause :(. Alternatively I would suggest you follow up pretty quickly. 

You probably know this already but I'm documenting this for posterity.

TL;DR

This is one bit of c++ that I don't like TBH. I think you'll get away with it because there is just one name for that function in that namespace today. inline is a hint to the compiler and it's not mandatory for compilers to honour that request. Heuristics , eh :) , 

inline functions in c++ headers end up being implemented as weak functions in the translation unit. And then if you have multiple translation units containing such weak functions, one ends up with a case where the linker will pick an arbitrary definition depending on what got included where. Normally this isn't a problem if the same *definition* reaches everywhere, however all programmers are human (well, still :)) and are capable of choosing names that are similar to what's been chosen here. 

From my reading these functions are in a namespace that is common with other parts of the code and thus with the right set of conditions we could end up with a situation where someone is debugging a problem with this undefined behaviour from C++11 as that's the dialect we program in. 

regards
Ramana

Reply:
Thanks, learnt something new today :)

Fair enough. I am adding `inline` for now in my added functions. We can create an Issue so that we ensure that we do it everywhere in the TVM codebase.